optuna/study/study.py

```diff
@@ -1045,14 +1067,15 @@ def _log_completed_trial(self, trial: trial_module.Trial, values: Sequence[float
             assert False, "Should not reach."
 
 
+@_convert_positional_args(n_positional_args=6)
```

Comment:
```suggestion
@_convert_positional_args(n_positional_args=7)
```

Reply:
Because `directions` is already a keyword-only argument, I think the number of arguments that can be specified by user is no more than 6.